x-pack/plugins/timelines/server/search_strategy/index_fields/index.ts

```diff
@@ -265,16 +267,26 @@ export const formatFirstFields = async (
 ): Promise<IndexField[]> => {
   return new Promise((resolve) => {
     setTimeout(() => {
+      const indexFieldNameHash: Record<string, number> = {};
       resolve(
         responsesFieldSpec.reduce(
-          (accumulator: IndexField[], fieldSpec: FieldSpec[], indexesAliasIdx: number) => {
-            missingFields.forEach((index) => {
-              const item = createFieldItem(beatFields, indexesAlias, index, indexesAliasIdx);
-              accumulator.push(item);
-            });
-            fieldSpec.forEach((index) => {
+          (accumulator: IndexField[], fieldSpec: FieldSpec[], indexesAliasId: number) => {
+            const indexesAliasIdx = responsesFieldSpec.length > 1 ? indexesAliasId : null;
+            [...missingFields, ...fieldSpec].forEach((index) => {
```

Comment:
You can use concat here as it will produce a new array and I _think_ it's slightly faster

Reply:
done!